lib/execution_engine2/utils/Condor.py

```diff
@@ -232,93 +119,69 @@ def _add_configurable_attributes(self, sub):
         return sub
 
     def _extract_resources_and_requirements(
-        self, sub: Dict[str, Any], cgrr: Dict[str, str]
-    ) -> Tuple[Dict[str, Any], str]:
+        self, sub: Dict[str, Any], job_reqs: JobRequirements
+    ) -> Dict[str, Any]:
         # Extract minimum condor resource requirements and client_group
-        resources = self.extract_resources(cgrr)
-        sub["request_cpus"] = resources.request_cpus
-        sub["request_memory"] = resources.request_memory
-        sub["request_disk"] = resources.request_disk
-        client_group = resources.client_group
+        sub["request_cpus"] = job_reqs.cpus
+        sub["request_memory"] = f"{job_reqs.memory_MB}MB"
+        sub["request_disk"] = f"{job_reqs.disk_GB}GB"
         # Set requirements statement
-        requirements = self._extract_requirements(cgrr=cgrr, client_group=client_group)
-        sub["requirements"] = " && ".join(requirements)
-        sub["+KB_CLIENTGROUP"] = f'"{client_group}"'
-        return (sub, client_group)
-
-    @staticmethod
-    def _modify_with_concierge(sub, concierge_params):
-        # Remove Concurrency Limits for this Job
-        del sub["Concurrency_Limits"]
-        # Override Clientgroup
-        sub["+KB_CLIENTGROUP"] = f'"{concierge_params.client_group}"'
-        if concierge_params.account_group:
-            sub["+AccountingGroup"] = concierge_params.account_group
-            # Override Resource Requirements
-        sub["request_cpus"] = concierge_params.request_cpus
-        sub["request_memory"] = concierge_params.request_memory
-        sub["request_disk"] = concierge_params.request_disk
-        # Build up requirements w/ custom requirements
-        sub["requirements"] = f'(CLIENTGROUP == "{concierge_params.client_group}")'
-        requirements = []
-        if concierge_params.requirements_list:
-            for item in concierge_params.requirements_list:
-                key, value = item.split("=")
-                requirements.append(f'({key} == "{value}")')
-        sub["requirements"] += " && ".join(requirements)
-
+        sub["requirements"] = self._create_requirements_statement(job_reqs)
+        sub["+KB_CLIENTGROUP"] = f'"{job_reqs.client_group}"'
         return sub
 
+    def _create_requirements_statement(self, job_reqs: JobRequirements):
```

Comment:
My position is that typing is optional for internal methods, and I usually add typing where I think it may be unclear what the expected thing is (e.g. here it's a specific object rather than a dict). Added the return type though.

Reply:
:+1: